### PR TITLE
MSC3861: `m.3pid_changes` capability should be disabled

### PR DESCRIPTION
**Pull request by MatMaul**
Fixes https://github.com/matrix-org/synapse/issues/16128.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by hughns** [comment]
FWIW, the approach that has been used elsewhere for MSC3861 is to error if the admin has tried to explicitly enable a feature that is not compatible with delegated auth. i.e. https://github.com/matrix-org/synapse/blob/develop/synapse/config/experimental.py#L151

I was intended to write a PR for this myself which did:

1. Change the default for `enable_3pid_changes` to be `False` if experimental MSC3861 is enabled: at https://github.com/matrix-org/synapse/blob/develop/synapse/config/registration.py#L136
2. Added a runtime check at https://github.com/matrix-org/synapse/blob/develop/synapse/config/experimental.py#L151
3. Unit tests to the runtime config error fires when misconfigured: https://github.com/matrix-org/synapse/blob/develop/tests/config/test_oauth_delegation.py#L56

**Review by clokep — DISMISSED** [review]
Per @hughns's review.

**Review by hughns — APPROVED** [review]
LGTM 👍

File: tests/config/test_oauth_delegation.py [hunk 1]
```diff
@@ -255,3 +255,8 @@ def test_session_lifetime_cannot_be_set(self) -> None:
         self.config_dict["session_lifetime"] = "24h"
         with self.assertRaises(ConfigError):
             self.parse_config()
+
+    def test_enable_3pid_changes_cannot_be_enabled(self) -> None:
+        self.config_dict.update(enable_3pid_changes=True)
```
Comment (clokep) [inline tests/config/test_oauth_delegation.py:260]:
```suggestion
        self.config_dict["enable_3pid_changes"] = True
```

**Comment by clokep** [comment]
I combined the news file with #16127.

**Review by clokep — APPROVED** [review]

**Pull request merged**